public/app/core/utils/ticks.ts

```diff
@@ -208,3 +211,34 @@ export function getStringPrecision(num: string): number {
     return num.length - dotIndex - 1;
   }
 }
+
+/**
+ * True if browser supports string locales for date parsing.
+ */
+function canUseLocaleDateStringLocales() {
+  try {
+    new Date().toLocaleDateString('i');
+  } catch (e) {
+    return e.name === 'RangeError';
+  }
+  return false;
+}
+
+/**
+ * Return time formatting string used for axis that display ticks with day granularity.
+ */
+function getMonthDayFormat() {
+  if (contextSrv.user.monthDayFormat && contextSrv.user.monthDayFormat !== 'browser') {
+    return contextSrv.user.monthDayFormat;
+  }
+
+  if (!canUseLocaleDateStringLocales()) {
+    return '%m/%d';
+  }
+
+  const language = navigator.languages ? navigator.languages[0] : navigator.language || 'en-US';
+  return new Date(Date.UTC(1977, 11, 20, 0, 0, 0))
+    .toLocaleDateString(language, { month: 'numeric', day: 'numeric' })
```

Comment:
Do you know of a concrete recommendation to use one over the other? Otherwise I see no basis to change this.

Reply:
No, i was just this thinking that it might be simpler just read the format directly instead of formating a date and parsing it to reverse engineer the format